pyface/data_view/abstract_value_type.py

```diff
@@ -0,0 +1,182 @@
+as text, color, icons, etc.
+
+It is up to the data view to take this standardized data and determine what
+and how to actually display it.
+"""
+
+from traits.api import ABCHasStrictTraits, Event, observe
+
+
+class AbstractValueType(ABCHasStrictTraits):
+    """ A value type converts raw data into data channels.
+
+    The data channels are editor value, text, color, image, and description.
+    The data channels are used by other parts of the code to produce the actual
+    display.
+
+    Subclasses should mark traits that potentially affect the display of values
+    with ``update=True`` metdadata, or alternatively fire the ``updated``
+    event when the state of the value type changes.
+    """
+
+    #: Fired when a change occurs that requires updating values.
+    updated = Event
+
+    def can_edit(self, model, row, column):
+        """ Return whether or not the value can be edited.
+
+        The default implementation is that cells that can be set are
+        editable.
+
+        Parameters
+        ----------
+        model : AbstractDataModel
+            The data model holding the data.
+        row : sequence of int
+            The row in the data model being queried.
+        column : sequence of int
+            The column in the data model being queried.
+
+        Returns
+        -------
+        can_edit : bool
+            Whether or not the value is editable.
+        """
+        return model.can_set_value(row, column)
+
+    def get_editable(self, model, row, column):
+        """ Return a value suitable for editing.
+
+        The default implementation is to return the underlying data value
+        directly from the data model.
+
+        Parameters
+        ----------
+        model : AbstractDataModel
+            The data model holding the data.
+        row : sequence of int
+            The row in the data model being queried.
+        column : sequence of int
+            The column in the data model being queried.
+
+        Returns
+        -------
+        value : any
+            The value to edit.
+        """
+        return model.get_value(row, column)
+
+    def set_editable(self, model, row, column, value):
+        """ Set a value that is returned from editing.
+
+        The default implementation is to set the value directly from the
+        data model.  Returns True if successful, False if it fails.
+
+        Parameters
+        ----------
+        model : AbstractDataModel
+            The data model holding the data.
+        row : sequence of int
+            The row in the data model being queried.
+        column : sequence of int
+            The column in the data model being queried.
+        value : any
+            The value to set.
+
+        Returns
+        -------
+        success : bool
+            Whether or not the value was successfully set.
+        """
+        if not self.can_edit(model, row, column):
+            return False
```

Comment:
Same comment as before.  It needs to be different from `set_value` on the model to avoid confusion.

I am leaning towards a nam-change of this, just not to 'get/set_value'.

Reply:
Maybe the ones on the model can be called `get_item`/`set_item` and then the ones here can be called `get_value` / `set_value`?

Reply:
Maybe.  I would like to somehow encapsulate the idea that this is the value that is used for _editing_.  In Qt this would be the "edit role" of the data.

Reply:
Maybe a bit verbose, but it might communicate responsibility better... how about `get_value_for_edit`? naming-thing-is-hard 😂